app/utils/eosio.js

```diff
@@ -368,29 +443,36 @@ class EosioService {
     chainId: process.env.SCATTER_CHAINID || this.node.chainID,
   });
 
-  handleCaseWithInvalidNode = async errorMsg => {
+  handleCaseWithInvalidNode = async (method, errorMsg, endpoint) => {
     try {
-      if (!this.isInvalidNodeHandling) {
-        this.isInvalidNodeHandling = true;
-
-        const { nodes, date } = this.getStoredNodes();
+      const { nodes, date } = this.getStoredNodes();
 
-        if (!nodes.length) throw new Error(errorMsg);
-
-        nodes[0].isInvalid = true;
+      if (!nodes.length && this.getDefaultEosConfig().endpoint === endpoint)
+        throw new Error(errorMsg);
 
+      if (nodes.find(x => x.endpoint === endpoint)) {
         localStorage.setItem(
           ENDPOINTS_LIST,
           JSON.stringify({
             date,
-            nodes,
+            nodes: nodes.filter(x => x.endpoint !== endpoint),
           }),
         );
+      }
 
+      const initializedWithScatter = this.eosApi
+        ? Boolean(!this.eosApi.signatureProvider)
+        : false;
+
+      if (initializedWithScatter) {
```

Comment:
Do we control the node selected in Scatter? I think we can not do anything if they use scatter.